src/hacking/building-for-openharmony.md

```diff
@@ -0,0 +1,120 @@
+2. The `hvigorw` build tool to compile apps into an app bundle and sign it.
+
+### Setting up the OpenHarmony SDK
+
+The OpenHarmony SDK is required to compile applications for OpenHarmony. 
+
+#### Downloading via DevEco Studio 
+
+[DevEco Studio] is an IDE for developing applications for HarmonyOS NEXT and OpenHarmony.
+It supports Windows and MacOS.
+You can manage installed OpenHarmony SDKs by clicking File->Settings and selecting "OpenHarmony SDK".
+After setting a suitable installation path, you can select the components you want to install for each available API version.
+DevEco Studio will automatically download and install the components for you.
+
+
+#### Manual installation of the OpenHarmony SDK (e.g. on Linux)
+
+<div class="warning _note">
+    Before rushing and downloading the OH SDK from gitee as described here, please note that you will also need `hvigor` to compile applications.
+    `hvigor` is currently only available via the [HarmonyOS NEXT commandline tools], which also contains a copy of the OpenHarmony SDK.
+</div>
+
+Go to the [OpenHarmony release notes] and select the version you want to compile for.
+Scroll down to the section "Acquiring Source Code from Mirrors" and click the download link for the version of "Public SDK package for the standard system" matching your host system.
+Extract the archive to a suitable location.
+Then switch into the SDK folder with `cd <sdk_folder>/<your_operating_system>` and unzip the zip files of the individual components.
+Preferably use the `unzip` command on the command-line, or manually ensure that the unzipped bundles are called e.g. `native` and not `native-linux-x64-5.x.y.z`.
+On windows, it is recommended to use 7zip to unzip the archives, since the windows explorer unzip tool is extremely slow.
+
+[DevEco Studio]: https://developer.huawei.com/consumer/cn/deveco-studio
+[OpenHarmony release notes]: https://gitee.com/openharmony/docs/tree/master/en/release-notes/
+[HarmonyOS NEXT commandline tools]: https://developer.huawei.com/consumer/cn/download/
+
+#### Manual installation of the HarmonyOS NEXT commandline tools
+
+The [HarmonyOS NEXT commandline tools] contain the OpenHarmony SDK and the following additional tools:
+
+- `codelinter` (linter)
+- `hstack` (crash dump stack analysis tool)
+- `hvigor` / `hvigorw` (build tool)
+- `ohpm` (package manager)
+
+Currently, the commandline tools package is not publicly available and requires a chinese Huawei account to download.
+
+#### Manual installation of hvigor without the commandline tools
+
+<div class="warning _note">
+This section is not fully tested and may change based on user feedback.
+It's recommended to install the commandline-tools bundle. If you decide to install manually, you need to take
+care to install the `hvigor` version matching the requirements of your project.
+</div>
+
+`hvigor` (not the wrapper `hvigorw`) is also available via `npm`. 
+1. Install the same nodejs version as the commandline-tools ship.
```

Comment:
Is there a way for users to figure out the node version? It might be helpful to add that here.

Reply:
I'm not sure. I thought I saw the required version documented somewhere, but I can't find it right now. 
So currently the only approach is "Check what the commandline tools bundle ships".  I did document the version shipped in Harmony OS NEXT though.
I imagine that a lot of the things will get easier after the official release, as more components become freely available / opensourced.